streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -40,6 +41,9 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
     @SuppressWarnings("unchecked")
     @Override
     public <T> List<T> stores(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+        if (streamThread.state() == StreamThread.State.DEAD) {
```

Comment:
That is exactly the thing we need to distinguish (ie, that is the state before this change) -- after a thread dies and the stores are migrated, the `StreamThreadStoreSupplier` of the died thread will still be checked if it can be queries -- thus, if we throw the whole attempt to query any store will fail, as a store is only queryable if all shards are queryable. Thus, a dead thread must not fail to make a store queryable.